src/interrupt.rs

```diff
@@ -28,20 +57,22 @@ pub unsafe fn enable() {
 
 /// Execute closure `f` in an interrupt-free context.
 /// This as also known as a "critical section".
-pub unsafe fn free<F>(f: F)
-    where F: FnOnce()
+pub unsafe fn free<F, R>(f: F) -> R
```

Comment:
This is "technically" a breaking change because it breaks generic code. But I expect that most people is using this in a concrete way: `interrupt::free(|| something())` and that will continue to work so I'm *not* going to bump the version to 0.2 because of this.